### PR TITLE
Upgrade hooks command improvement

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refacto the Symfony command to update hooks on autoupgrade module, find old and new hooks based on the fixtures files and then export the two SQL queries in the upgrade file
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI green UI tests green Maybe QA by dev
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/11368580096
| Fixed issue or discussion?     | ~
| Related PRs       | ~
| Sponsor company   | ~

### Description

This command was updated to use the XML fixture files as a reference to detect new and obsolete hooks, it can then update the SQL files in the autoupgrade module.

The command was used to generated this PR https://github.com/PrestaShop/autoupgrade/pull/957
The CLI command used to generate it was:
```sh
./bin/console prestashop:update:sql-upgrade-file-hooks-listing 8.2.0 /path/to/module/autoupgrade
```

### CODE REVIEW

**Review by tleon — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]
Does it require a rebase on `9.0.x`?

**Comment by jolelievre** [comment]
> Does it require a rebase on `9.0.x`?

Absolutely

**Review by nicosomb — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
